Source/Dafny/Verifier/Translator.cs

```diff
@@ -10564,7 +10564,9 @@ void TrStmt(Statement stmt, BoogieStmtListBuilder builder, List<Variable> locals
           if (assertStmt != null && assertStmt.Proof != null) {
             proofBuilder = new BoogieStmtListBuilder(this);
             AddComment(proofBuilder, stmt, "assert statement proof");
+            CurrentIdGenerator.Push();
```

Comment:
Could we add a little helper method to `FreshIdGenerator` to better structure these paired calls?

```C#
public void WithNewScope(Action action) {
  Push();
  action();
  Pop();
}
```

Then each time we open a scope is visually more obvious in the source. E.g.:

```C#
CurrentIdGenerator.WithNewScope(() => {
  TrStmt(((AssertStmt)stmt).Proof, proofBuilder, locals, etran);
});
```

I recall doing something similar in one of my previous PRs but can't seem to find it now...

Reply:
Yeah, I can add that but that'd be a refactoring ,and if we're refactoring I'd prefer removing the need for Push and Pop calls altogether. There's already code that specifies where Dafny scopes are created in another location, for example [here](https://github.com/dafny-lang/dafny/blob/master/Source/Dafny/Resolver.cs#L11053), and specifying it in two locations means the two can go out of sync.

Reply:
Definitely not a blocker. I mainly just want the technique to be in our heads in general - I think it can help structure the code more clearly sometimes.